src/Discord/Internal/Types/Guild.hs

```diff
@@ -276,78 +274,16 @@ instance FromJSON Role where
          <*> o .: "color"
          <*> o .: "hoist"
          <*> o .: "position"
-         <*> o .: "permissions"
+         <*> fmap (readMaybe . T.unpack) (o .: "permissions") .!= RolePermissions 0
```

Comment:
the use of readmaybe here indicates that we may want a FromJSON instance for RolePermissions. Further, if we can't read the string they throw at us, we should fail, so the use of readMaybe here isn't advised.